docs/blog/2021-09-27-crash.md

```diff
@@ -0,0 +1,328 @@
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+```
+
+For the record, this is the first internal error I was able to uncover so far. Given this success with categorical strings, it was natural to look if `one_of` could be similarly broken, but the generator just chose the one variant despite its `0.0` weight.
+
+\#10 Unsupported types on import
+
+Databases can sometimes contain strange things, and so far the support is in beta, so it was expected that I would find types for which we currently don't implement import. This includes JSON for mysql and postgres, the mysql spatial datatypes as well as postgres' geometric types, user-defined enumerations, postgres' network address types, postgres arrays (soon only nested ones), etc.
+
+The way to reproduce that is to create a table with a field of the type, e.g. here with mysql
+
+```sql
+CREATE TABLE IF NOT EXISTS json (
+    data JSON
+);
+
+DELETE FROM json;
+
+INSERT INTO json (data) VALUES ('{ "a": ["b", 42] }');
+```
+
+Now call `synth import fail --from mysql://<user>:<password>@<host>:<port>/<database>` to get
+
+```console
+Error: We haven't implemented a converter for json
+```
+
+Since the error is mostly the same for all types, and was somewhat expected, I won't claim a point for each type here.
+
+\#11 Exporting an array of nulls into postgres is not correctly implemented, so
+
+```json synth
+{
+    "type": "array",
+    "length": {
+        "type": "number",
+        "subtype": "u32",
+    },
+    "content": {
+        "type": "object",
+        "s": {
+            "type": "array",
+            "length": 1,
+            "content": {
+                "type": "null"
+            }
+        }
+    }
+}
+```
+
+will give us a `wrong data type` error from postgres. The problem here is that we lose the type information from the generator, and just emit `null` values which do not allow us to construct the right types for encoding into a postgres buffer. The solution would be to re-architect the whole system to reinstate that type information, possibly side-stepping sqlx in the process. Note that this is not equal to [issue #171](https://github.com/getsynth/synth/issues/171), which relates to nested arrays.
+
+\#12 going back to \#3, I thought about other ways to make the code overconsume resources. But time and memory are only one thing to consume, in fact it's easy enough to consume another: The stack. The following bash script:
+
+```bash
+X='{ "type": "null" }'
+
+for i in $(seq 0 4096)
+do
+    X="{ \"type\": \"string\", \"format\": { \"format\": \"{x}\", \"arguments\": { \"x\": $X } } }"
+done
+
+echo $X > 12_stack_depth/stack_depth.json
+synth gen --size 1 12_stack_depth
+```
+
+will generate the following error:
+
+```console
+Error: Unable to open the namespace
+
+Caused by:
+    0: at file 12_stack_depth/stack_depth.json
+    1: Failed to parse collection
+    2: recursion limit exceeded at line 1 column 2929
+```
```

Comment:
hah! that's cool :D

Reply:
I knew I had to overflow the stack at least once.